### PR TITLE
CAMEL-13631: PropertyBindingSupport to support key filtering and transformation

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

**Comment by lburgazzoli** [comment]
@davsclaus I've tried to add a generic solution for CAMEL-13631, but let me know if you feel it is over complicated/engineered.

**Comment by lburgazzoli** [comment]
@davsclaus yeah, I started with something simple but get more complicated to have a generic solution and I didn't like it too much too, in any case I've simplified it to be similar to what we have in IntrospectionSupport.

File: core/camel-support/src/main/java/org/apache/camel/support/PropertyBindingSupport.java [hunk 1]
```diff
@@ -287,9 +323,22 @@ public static boolean bindProperties(CamelContext camelContext, Object target, M
...
             Map.Entry<String, Object> entry = iter.next();
-            if (bindProperty(camelContext, target, entry.getKey(), entry.getValue(), nesting, deepNesting, fluentBuilder, reference, placeholder)) {
-                iter.remove();
-                rc = true;
+            String key = entry.getKey();
+            Object value = entry.getValue();
+
+            if (isNotEmpty(optionPrefix)) {
+                if (!key.startsWith(optionPrefix)) {
+                    return false;
```
Comment (davsclaus) [inline core/camel-support/src/main/java/org/apache/camel/support/PropertyBindingSupport.java]: I think this is wrong, we should just skip keys without the prefix, and not return

File: core/camel-support/src/main/java/org/apache/camel/support/PropertyBindingSupport.java [hunk 2]
```diff
@@ -16,6 +16,9 @@
  */
 package org.apache.camel.support;
 
+import static org.apache.camel.support.IntrospectionSupport.findSetterMethods;
```
Comment (davsclaus) [inline core/camel-support/src/main/java/org/apache/camel/support/PropertyBindingSupport.java]: Checkstyle issues here

Comment (davsclaus) [reply core/camel-support/src/main/java/org/apache/camel/support/PropertyBindingSupport.java]: As the import order is changed now

File: core/camel-support/src/main/java/org/apache/camel/support/PropertyBindingSupport.java [hunk 3]
```diff
@@ -258,11 +257,25 @@ private static boolean doAutowireSingletonPropertiesFromRegistry(CamelContext ca
...
 
     /**
-     * Binds the properties to the target object, and removes the property that was bound from properties.
+     * Binds the properties with the given prefix to the target object, and removes the property that was bound from properties.
+     * Note that the prefix is removed from the key before the property is bound.
+     *
+     * @param camelContext  the camel context
+     * @param target        the target object
+     * @param properties    the properties where the bound properties will be removed from
+     * @param optionPrefix  the prefix used to filter properties
```
Comment (davsclaus) [inline core/camel-support/src/main/java/org/apache/camel/support/PropertyBindingSupport.java]: Also add the optionPrefix to the Builder (see top of class) so you can use this style as well

File: core/camel-support/src/main/java/org/apache/camel/support/PropertyBindingSupport.java (hunk 1 above)
Comment (lburgazzoli) [reply core/camel-support/src/main/java/org/apache/camel/support/PropertyBindingSupport.java]: oh yes, taken fomr the introspection support, thx for spotting

**Comment by lburgazzoli** [comment]
@davsclaus should be better now

**Pull request merged**